ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/MessageListContainerFragmentTest.java

```diff
@@ -0,0 +1,403 @@
+    PowerMockito.mockStatic(MessengerPref.class);
+    PowerMockito.mockStatic(Kayako.class);
+    PowerMockito.mockStatic(MessageListContainerFactory.class);
+    PowerMockito.mockStatic(FileAttachmentUtil.class);
+    PowerMockito.mockStatic(Toast.class);
+    PowerMockito.mockStatic(Gravity.class);
+    Mockito.when(messageListContainerFragment.getActivity()).thenReturn(fragmentActivity);
+    Mockito.when(messageListContainerFragment.getContext()).thenReturn(context);
+    Mockito.when(fragmentActivity.getIntent()).thenReturn(intent);
+    Mockito.when(intent.getExtras()).thenReturn(bundle);
+  }
+
+  @Test
+  public void callRetainInstanceOnCreate() {
+    // Arrange
+    PowerMockito.when(MessageListContainerFactory.getData())
+        .thenReturn(contractData);
+
+    // Act
+    messageListContainerFragment.onCreate(bundle);
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).onCreate(bundle);
+    Mockito.verify(messageListContainerFragment).setRetainInstance(true);
+    PowerMockito.verifyStatic(MessageListContainerFactory.class, Mockito.times(1));
+    MessageListContainerFactory.getPresenter(messageListContainerFragment);
+  }
+
+  @Test
+  public void callInflateOnCreateView() {
+    // Arrange
+
+    // Act
+    messageListContainerFragment.onCreateView(inflater, viewGroup, null);
+
+    // Assert
+    Mockito.verify(inflater).inflate(R.layout.ko__fragment_message_list_container, viewGroup, SAVED_INSTANCE_STATE);
+  }
+
+  @Test
+  public void fillFieldsOnViewCreated() {
+    // Arrange
+
+    // Act
+    messageListContainerFragment.onViewCreated(view, bundle);
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).onViewCreated(view, bundle);
+    Mockito.verify(messageListContainerFragment, Mockito.times(3)).getChildFragmentManager();
+  }
+
+  @Test
+  public void callOnActivityCreatedWithNonNullBundle() {
+    // Arrange
+    MessageListContainerFragment fragmentMock = PowerMockito.mock(MessageListContainerFragment.class);
+    bundle.putLong(KayakoSelectConversationActivity.ARG_CONVERSATION_ID, 1L);
+    Mockito.when(fragmentMock.getActivity()).thenReturn(fragmentActivity);
+
+    // Act
+    fragmentMock.onActivityCreated(bundle);
+
+    // Assert
+    Mockito.verify(fragmentMock).onActivityCreated(bundle);
+  }
+
+  @Test
+  public void testOnActivityResultWithAddAttachment() {
+    // Arrange
+    int addAttachmentCode = Whitebox.getInternalState(MessageListContainerFragment.class, "REQUEST_CODE_ADD_ATTACHMENT");
+
+    // Act
+    messageListContainerFragment.onActivityResult(addAttachmentCode, RESULT_CODE_OK, intent);
+
+    // Assert
+    PowerMockito.verifyStatic(FileAttachmentUtil.class);
+    FileAttachmentUtil.getFileOnActivityResult(RESULT_CODE_OK, intent);
+  }
+
+  @Test
+  public void testOnActivityResultWithViewAttachment() {
+    // Arrange
+    int viewAttachmentCode = Whitebox.getInternalState(MessageListContainerFragment.class, "REQUEST_CODE_VIEW_ATTACHMENT_BEFORE_SENDING");
```

Comment:
⚠️  A line of code should not exceed 120 characters in order to be readable and avoid horizontal scroll. Consider using the 'line-wrapping' technique to format this line of code in multiple shorter lines. The formula for wrapping can be found at the below link.  

**[More details](https://confluence.devfactory.com/x/wgbhEw)**
          
## The possible solution in this case

```JAVA
     int viewAttachmentCode = Whitebox.getInternalState(MessageListContainerFragment.class,
                                                       "REQUEST_CODE_VIEW_ATTACHMENT_BEFORE_SENDING");

```
 

<a id="ef242750-4f1d-11e8-b712-a55fdd633c3d" href="none" codereview-uuid-reserved-tag />